pkg/process/config/yaml_config.go

```diff
@@ -270,3 +274,22 @@ func (a *AgentConfig) setCheckInterval(ns, check, checkKey string) {
 		a.CheckIntervals[checkKey] = time.Duration(interval) * time.Second
 	}
 }
+
+// Separate handler for initializing the process discovery check.
+// Since it has its own unique object, we need to handle loading in the check config differently separately
+// from the other checks.
+func (a *AgentConfig) initProcessDiscoveryCheck() {
+	root := key(ns, "process_discovery")
+
+	// We don't need to check if the key exists since we already bound it to a default in InitConfig
+	a.CheckIntervals[DiscoveryCheckName] = config.Datadog.GetDuration(key(root, "interval"))
+
+	// Discovery check should be only enabled when process_config.process_discovery.enabled = true and
+	// process_config.enabled is set to "false". This effectively makes sure the check only runs when other checks are
+	// disabled, while also respecting the users wishes when they want to disable either the check or the process agent completely.
+	processAgentEnabled := strings.ToLower(config.Datadog.GetString(key(ns, "enabled")))
```

Comment:
Curious why we need `ToLower` here.

Reply:
I added it as an extra layer of sanitization for inputs to the yaml file. I suppose we don't need it, but it's an extra precaution in case someone types `enabled: False`.

Let me know what your thoughts are!